transport/src/main/java/io/netty/channel/socket/nio/NioChannelOption.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.socket.nio;
+
+import io.netty.channel.ChannelException;
+import io.netty.channel.ChannelOption;
+
+import java.io.IOException;
+import java.net.SocketOption;
+import java.nio.channels.NetworkChannel;
+import java.util.Set;
+
+/**
+ * Provides {@link ChannelOption} over a given {@link SocketOption} which is then passed through the underlying
+ * {@link NetworkChannel}.
+ */
+public final class NioChannelOption<T> extends ChannelOption<T> {
+
+    private final SocketOption<T> option;
+
+    @SuppressWarnings("deprecation")
+    private NioChannelOption(SocketOption<T> option) {
+        super(option.name());
+        this.option = option;
+    }
+
+    /**
+     * Returns a {@link ChannelOption} for the given {@link SocketOption}.
+     */
+    public static <T> ChannelOption<T> of(SocketOption<T> option) {
+        return new NioChannelOption<T>(option);
+    }
+
+    // Internal helper methods to remove code duplication between Nio*Channel implementations.
+    static <T> boolean setOption(NetworkChannel channel, NioChannelOption<T> option, T value) {
+        try {
+            channel.setOption(option.option, value);
+            return true;
+        } catch (UnsupportedOperationException ignore) {
```

Comment:
Did you consider checking supportedOptions instead of catching the exception?
```
if (!channel.supportedOptions().contains(option.value)) {
  return false;
}
```

Given the design of the API, I'd expect every implementation to have `supportedOptions()` be a simple getter. This is at least true for all the implementations in the JDK.